flow/cmd/validate_mirror.go

```diff
@@ -190,3 +196,93 @@ func (h *FlowRequestHandler) CheckIfMirrorNameExists(ctx context.Context, mirror
 
 	return nameExists.Bool, nil
 }
+
+func (h *FlowRequestHandler) ValidateTableAdditions(ctx context.Context, req *protos.ValidateTableAdditionsRequest) (*protos.ValidateCDCMirrorResponse, error) {
+	sourcePeer, err := connectors.LoadPeer(ctx, h.pool, req.SourcePeerName)
+	if err != nil {
+		slog.Error("table addition validation: failed to load source peer", slog.String("peer", req.SourcePeerName))
+		return nil, err
+	}
+
+	sourcePeerConfig := sourcePeer.GetPostgresConfig()
+	if sourcePeerConfig == nil {
+		slog.Error("table addition validation: source peer config is not postgres", slog.String("peer", req.SourcePeerName))
+		return nil, errors.New("source peer config is not postgres")
+	}
+
+	pgPeer, err := connpostgres.NewPostgresConnector(ctx, nil, sourcePeerConfig)
+	if err != nil {
+		displayErr := fmt.Errorf("failed to create postgres connector: %v", err)
+		h.alerter.LogNonFlowWarning(ctx, telemetry.EditMirror, req.FlowJobName, displayErr.Error())
+		return nil, displayErr
+	}
+	defer pgPeer.Close()
+
+	srcAddedTableNames := make([]string, 0, len(req.AddedTables))
+	addedTableValues := make([]string, 0, len(req.AddedTables))
+	for _, tableMapping := range req.AddedTables {
+		parsedTable, parseErr := utils.ParseSchemaTable(tableMapping.SourceTableIdentifier)
+		if parseErr != nil {
+			displayErr := fmt.Errorf("invalid source table identifier: %s", parseErr)
+			h.alerter.LogNonFlowWarning(ctx, telemetry.EditMirror, req.FlowJobName, displayErr.Error())
+			return nil, displayErr
+		}
+
+		addedTableValues = append(addedTableValues, fmt.Sprintf(`(%s::text,%s::text)`,
+			connpostgres.QuoteLiteral(parsedTable.Schema),
+			connpostgres.QuoteLiteral(parsedTable.Table),
+		))
+
+		srcAddedTableNames = append(srcAddedTableNames, tableMapping.SourceTableIdentifier)
+	}
+
+	if req.PublicationName != "" {
+		publicationErr := pgPeer.CheckIfTablesAreInPublication(ctx, req.PublicationName, addedTableValues)
+		if publicationErr != nil {
+			h.alerter.LogNonFlowWarning(ctx, telemetry.EditMirror, req.FlowJobName, publicationErr.Error())
+			return nil, publicationErr
+		}
```

Comment:
```suggestion
		if err := pgPeer.CheckIfTablesAreInPublication(ctx, req.PublicationName, addedTableValues); err != nil {
			h.alerter.LogNonFlowWarning(ctx, telemetry.EditMirror, req.FlowJobName, err.Error())
			return nil, err
		}
```
I know it seems nice to avoid shadowing, but what happens is someone makes a change to the error message or something, assumes the variable is called `err`, & ends up having their code reference some always-nil `err` from earlier in the code